weekly_quiz/admin.py

```
@@ -0,0 +1,146 @@
+        also checks if the answer has been submitted and Foreign key has
+        been included
+        """
+        super(WeeklyQuizAnswerFormset, self).clean()
+        char_limit_answer = len(self.instance.question)
+        char_limit_response = 0
+        for form in self.forms:
+            if not hasattr(form, 'cleaned_data'):
+                continue
+
+            if ("answer" not in form.cleaned_data):
+                raise forms.ValidationError("You need to complete the answers")
+
+            else:
+                char_limit_answer = char_limit_answer + len(form.cleaned_data['answer'])
+
+                if char_limit_answer > 160:
+                    raise forms.ValidationError("You have gone beyond the"
+                                                " character limit"
+                                                " please shorten questions"
+                                                " and/or answers")
+
+            if ("response" not in form.cleaned_data):
+                raise forms.ValidationError("You need to complete the responses")
+
+            else:
+                char_limit_response = char_limit_response + len(form.cleaned_data['response'])
+                if char_limit_response > 160:
+                    raise forms.ValidationError("You have gone beyond the"
+                                                " character limit"
+                                                " please shorten the responses")
+
+        try:
+        # Checking if Foreign ID has been included otherwise gives
+        # an DoesNotExist exception at which point it skips
+            query = (WeeklyQuizQuestion.objects.all().
+                     filter(quiz_id=self.instance.quiz_id))
+            if len(query) >= 2:
+                query = (WeeklyQuizId.objects.
+                         get(pk=self.instance.quiz_id.id))
+                query.completed = True
+                query.save()
+        except:
```

Please capture the exact Exception you're expecting, silently catching everything is too broad.